### PR TITLE
Create xml in to_xml if options present

### PR DESCRIPTION
**Pull request by cplehm**
to_xml should return formatted xml if options is present. Worked on in conjunction with @vignesh-vs-in. 

Please review, @jahfer @stephenminded

### CODE REVIEW

File: spec/money_spec.rb [hunk 1]
```diff
@@ -157,8 +158,14 @@
...
   end
 
-  it "returns cents in as_xml" do
-    expect(Money.new(1.00).as_xml).to eq("1.00")
+  it "returns xml in to_xml with options present" do
+    options = {  
+      :root=>"price",
+      :format=>"xml",
+      :builder=>Builder::XmlMarkup.new,
+      :skip_instruct=>true
```
Comment (stephenminded) [inline spec/money_spec.rb]: Can you use the new Ruby hash syntax throughout this PR? IE: `skip_instruct: true`.

File: lib/money/money.rb [hunk 2]
```diff
@@ -138,11 +138,11 @@ def as_json(*args)
   end
 
   def to_xml(options = {})
-    to_s
-  end
-
-  def as_xml(*args)
-    to_s
+    if options.try(:[], :builder)
+      options[:builder].tag!(options[:root], to_s, {:type=>"decimal"})
```
Comment (stephenminded) [inline lib/money/money.rb]: `options[:root]` is required here (unless you want to provide a default, but I can't think of a sane way to do that). So, can you use `options.fetch(:root)` which will raise a `KeyError` if it isn't provided.

**Comment by vignesh-vs-in** [comment]
Fixed the ruby hashing format. Can we get a go ahead for this, if everything else is looking fine?

File: lib/money/money.rb [hunk 3]
```diff
@@ -138,11 +138,11 @@ def as_json(*args)
   end
 
   def to_xml(options = {})
-    to_s
-  end
-
-  def as_xml(*args)
-    to_s
+    if options.try(:[], :builder)
+      options[:builder].tag!(options.fetch(:root), to_s, {:type=>"decimal"})
```
Comment (stephenminded) [inline lib/money/money.rb]: Use the new hash syntax here as well `type: "decimal"` (also, no `{}` needed)

**Comment by stephenminded** [comment]
One comment, after that 🚢

File: lib/money/money.rb [hunk 4]
```diff
@@ -138,11 +138,11 @@ def as_json(*args)
   end
 
   def to_xml(options = {})
-    to_s
-  end
-
-  def as_xml(*args)
-    to_s
+    if options.try(:[], :builder)
```
Comment (jahfer) [inline lib/money/money.rb]: `if options.key?(:builder)`

**Comment by jahfer** [comment]
don't know much about the XML path of things, but LGTM 👍

**Comment by jeromecornet** [comment]
Am I reading the code correctly in that it is serializing every money object under a <price> tag now?

**Comment by stephenminded** [comment]
> Am I reading the code correctly in that it is serializing every money object under a <price> tag now?

If the `builder` and `root` options are provided, it will serialize the money object under the tag specified by `root`. Otherwise we'll serialize the money value as a strait up string with no tag. I'm unsure if rails provides these options by default when serializing to XML, however.

**Comment by vignesh-vs-in** [comment]
Rails expects a serialized xml tag when it provides the serialization options. It does provide the options when serializing a model.

**Pull request merged**